tests/ui/invalid_result_conversion.stderr

```diff
@@ -1,14 +1,12 @@
-error[E0277]: the trait bound `Result<(), MyError>: IntoPyCallbackOutput<_>` is not satisfied
-   --> tests/ui/invalid_result_conversion.rs:21:1
-    |
-21  | #[pyfunction]
-    | ^^^^^^^^^^^^^ the trait `IntoPyCallbackOutput<_>` is not implemented for `Result<(), MyError>`
-    |
-    = help: the following implementations were found:
-              <Result<T, E> as IntoPyCallbackOutput<U>>
-note: required by a bound in `pyo3::callback::convert`
-   --> src/callback.rs
-    |
-    |     T: IntoPyCallbackOutput<U>,
-    |        ^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `pyo3::callback::convert`
-    = note: this error originates in the attribute macro `pyfunction` (in Nightly builds, run with -Z macro-backtrace for more info)
+error[E0599]: the method `into_py_result` exists for enum `Result<(), MyError>`, but its trait bounds were not satisfied
+  --> tests/ui/invalid_result_conversion.rs:21:1
+   |
+21 | #[pyfunction]
+   | ^^^^^^^^^^^^^ method cannot be called on `Result<(), MyError>` due to unsatisfied trait bounds
+   |
+   = note: the following trait bounds were not satisfied:
+           `&Result<(), MyError>: IntoPy<Py<PyAny>>`
+           which is required by `&Result<(), MyError>: IntoPyResult<Result<&Result<(), MyError>, PyErr>>`
+           `&mut Result<(), MyError>: IntoPy<Py<PyAny>>`
+           which is required by `&mut Result<(), MyError>: IntoPyResult<Result<&mut Result<(), MyError>, PyErr>>`
```

Comment:
Yeah I think that having `IntoPyResult<T>` as suggested above would make the errors here even nicer, because it'd remove one level of `Result<_, PyErr>` from these messages.